src/machine.rs

```diff
@@ -103,7 +104,7 @@ impl<'tcx> Frame<'tcx> {
 #[derive(Debug)]
 pub enum Address {
     Heap(usize), // pointer to offset in memory.
-    Local(Local)
+    Local(Local, usize),
```

Comment:
Am I being daft, or can I not see the change to the `Local` signature in this PR?

Reply:
Not sure what you mean? Could you be confusing MIR `Local`s with Mirkat's `Address::Local`?

Reply:
Somewhere you've added one argument to a thing called `Local`, but I don't see a change to it's definition.

Reply:
I assumed (but I might be wrong) that this is the thing that's changed in rust nightly?

Reply:
This `usize` corresponds to the offset inside the byte vector stored at that local index. This will be used to read a specific field of an aggregate type instead of pulling the whole thing into memory. I haven't pushed the code that does this yet as part of this PR, as I think it deserves its own PR and the fix applied here is minimal in making the simple tuple assignment test pass.

Reply:
So it's either a Rustc change you are responding to (as Laurie says) or this code doesn't compile. The former?

Reply:
I think I'm misunderstanding your question. This is not a change to Rustc's `Local` signature. My `Address` enum now needs to store the offset as part of a `Local` variant as projection evaluations are now required as part of tuple creation. 

`Address::Local` and `Mir::Local` are orthogonal. I admit it's a bit confusing as they use the same names, perhaps I should change this.

Reply:
Ah, I understand. I had actually misread. My bad.